### PR TITLE
feat: implement document origin screenshots

### PR DESCRIPTION
**Pull request by jrandolf**

### CODE REVIEW

File: tests/browsing_context/capture_screenshot/test_capture_screenshot.py [hunk 1]
```diff
@@ -160,3 +160,48 @@ async def test_screenshot_oopif(websocket, context_id, html, iframe,
...
         assert_images_similar(resp["result"]["data"], png_base64)
+
+
+@pytest.mark.asyncio
+async def test_screenshot_document(websocket, context_id, query_selector,
+                                  get_cdp_session_id, html):
+    await goto_url(websocket, context_id, html('<div style="width: 100px; height: 100px; background: red"></div>'))
+    session_id = await get_cdp_session_id(context_id)
+
+    # Set a fixed viewport to make the test deterministic.
```
Comment (OrKoN) [inline tests/browsing_context/capture_screenshot/test_capture_screenshot.py:171]: use `browsingContext.setViewport`?

File: tests/browsing_context/capture_screenshot/test_capture_screenshot.py [hunk 2]
```diff
@@ -160,3 +144,43 @@ async def test_screenshot_oopif(websocket, context_id, html, iframe,
...
+async def test_screenshot_document(websocket, context_id, query_selector, html):
+    await goto_url(websocket, context_id, html('<div style="width: 100px; height: 100px; background: red"></div>'))
+
+    # Set a fixed viewport to make the test deterministic.
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.setViewport",
+            "params": {
+                "context": context_id,
+                "viewport": {
```
Comment (OrKoN) [inline tests/browsing_context/capture_screenshot/test_capture_screenshot.py:159]:
```suggestion
                "devicePixelRatio": 1,
                "viewport": {
```

File: tests/browsing_context/capture_screenshot/test_capture_screenshot.py [hunk 3]
```diff
@@ -125,21 +114,16 @@ async def test_screenshot_oopif(websocket, context_id, html, iframe,
...
-    await execute_command(
+    await send_JSON_command(
         websocket, {
-            "method": "cdp.sendCommand",
+            "method": "browsingContext.setViewport",
             "params": {
-                "method": "Emulation.setDeviceMetricsOverride",
-                "params": {
+                "context": context_id,
+                "viewport": {
```
Comment (OrKoN) [inline tests/browsing_context/capture_screenshot/test_capture_screenshot.py:123]:
```suggestion
                "devicePixelRatio": 1,
                "viewport": {
```

File: tests/browsing_context/capture_screenshot/test_capture_screenshot.py [hunk 4]
```diff
@@ -69,24 +65,20 @@ async def test_screenshot(websocket, context_id, png_filename,
...
                     "width": 200,
                     "height": 200,
-                    "deviceScaleFactor": 1.0,
-                    "mobile": False,
                 },
-                "session": session_id
+                "devicePixelRatio": 1,
```
Comment (thiagowfx) [inline tests/browsing_context/capture_screenshot/test_capture_screenshot.py:81]: Please set these to 1.0 instead as they are floats

**Review by thiagowfx — APPROVED** [review]

**Review by thiagowfx — APPROVED** [review]

**Pull request merged**